op-chain-ops/state/memory_db.go

```diff
@@ -165,6 +168,8 @@ func (db *MemoryStateDB) SetCode(addr common.Address, code []byte) {
 	db.rw.Lock()
 	defer db.rw.Unlock()
 
+	db.createAccount(addr)
```

Comment:
this matches what the geth statedb does: https://github.com/ethereum-optimism/op-geth/blob/ed8e9f54ed0a5069e77ea34895ff12a9ac4ca17f/core/state/statedb.go#L406

we can probably update a bunch of these methods to create an account lazily